core/src/test/java/io/questdb/test/griffin/SqlOptimiserTest.java

```diff
@@ -149,6 +149,74 @@ public void testAliasAppearsInFuncArgs4() throws Exception {
         });
     }
 
+    @Test
+    public void testQueryPlanForLastAggregateFunctionOnDesignatedTimestampColumn() throws Exception {
```

Comment:
Please add more tests for more tricky cases: e.g. aliases, explicit GROUP BY clause, and nested queries with TIMESTAMP clause (`select last(non_designated_ts) from ((select * from x order by non_designated_ts) timestamp(non_designated_ts))`). It's important to check that the rewrite doesn't lead to regressions.

Reply:
(select last(non_designated_ts) from ((select * from x order by non_designated_ts) timestamp(non_designated_ts)))

@puzpuzpuz for this the expected plan should be group by workers and forward scan right? as we are not handling dynamically allocated designated timestamp column here

Reply:
`SELECT *` in subquery in query 1 above is redundant as the outer query only needs the `non_designated_ts` column. That's why query engine rewrites query 1 to query 2 internally.